distributed/comm/tests/test_ucx.py

```diff
@@ -350,3 +354,102 @@ def test_ucx_deserialize():
 async def test_ucx_protocol(cleanup, port):
     async with Scheduler(protocol="ucx", port=port) as s:
         assert s.address.startswith("ucx://")
+
+
+@pytest.mark.asyncio
+async def test_ucx_config(cleanup):
+    ucx = {"nvlink": True, "infiniband": True, "tcp-over-ucx": True}
+
+    def get_ucp_config():
+        import ucp
+
+        return ucp.get_config()
+
+    with dask.config.set(ucx=ucx):
+        async with Scheduler(protocol="ucx") as s:
+            async with Worker(s.address) as a:
+                async with Client(s.address, asynchronous=True) as c:
+
+                    # scheduler is configured with NVLINK/IB
+                    ucx_config = await c.run_on_scheduler(get_ucp_config)
+                    assert ucx_config.get("TLS") == "rc,tcp,sockcm,cuda_copy,cuda_ipc"
+                    assert ucx_config.get("SOCKADDR_TLS_PRIORITY") == "sockcm"
+
+                    # worker is configured with NVLINK/IB
+                    worker_ucx_config = await c.run(get_ucp_config)
+                    ucx_config = worker_ucx_config[a.contact_address]
+                    assert ucx_config.get("TLS") == "rc,tcp,sockcm,cuda_copy,cuda_ipc"
+                    assert ucx_config.get("SOCKADDR_TLS_PRIORITY") == "sockcm"
+
+
+def test_ucx_config_env_var(cleanup, loop, monkeypatch):
+    def get_ucp_config():
+        import ucp
+
+        return ucp.get_config()
+
+    monkeypatch.setenv("DASK_UCX__NVLNK", "False")
+    monkeypatch.setenv("DASK_UCX__INFINIBAND", "True")
+    monkeypatch.setenv("DASK_UCX__TCP_OVER_UCX", "True")
+
+    port = "13337"
+    sched_addr = "ucx://%s:%s" % (get_ip(), port)
+    with popen(
+        ["dask-scheduler", "--no-dashboard", "--protocol", "ucx", "--port", port]
+    ) as sched:
+        with popen(["dask-worker", sched_addr, "--no-dashboard", "--protocol", "ucx"]):
+            with Client(sched_addr, loop=loop, timeout=10) as c:
+                while not c.scheduler_info()["workers"]:
+                    sleep(0.1)
+
+                # scheduler is configured with NVLINK (no cuda_ipc)
+                ucx_config = c.run_on_scheduler(get_ucp_config)
+                assert ucx_config.get("TLS") == "rc,tcp,sockcm,cuda_copy"
+                assert ucx_config.get("SOCKADDR_TLS_PRIORITY") == "sockcm"
+
+                # worker is configured with NVLINK (no cuda_ipc)
+                ucx_config = c.run(get_ucp_config)
+                assert ucx_config.get("TLS") == "rc,tcp,sockcm,cuda_copy"
+                assert ucx_config.get("SOCKADDR_TLS_PRIORITY") == "sockcm"
+
+
+@pytest.mark.asyncio
+async def test_ucx_w_rmm(cleanup):
+    ucx = {"nvlink": True, "infiniband": True, "tcp-over-ucx": True}
+    size = "1000.00 MB"
+    rmm = {"pool-size": size}
+
+    def get_ucp_config():
+        import ucp
+
+        return ucp.get_config()
+
+    def get_rmm():
+        import rmm
+
+        return rmm.get_info()
```

Comment:
Maybe we can replace these functions with just  `rmm.get_info` and `ucp.get_config`?

Reply:
:). Definitely!